GeneratorInterface/RivetInterface/plugins/GenParticles2HepMCConverter.cc

```diff
@@ -109,27 +111,38 @@ void GenParticles2HepMCConverter::produce(edm::Event& event, const edm::EventSet
 
     hepmc_particles.push_back(hepmc_particle);
     genCandToHepMCMap[p] = hepmc_particle;
+
+    // Find incident proton pair
+    if ( p->mother() == nullptr and std::abs(p->eta()) > 5 and std::abs(p->pz()) > 1000 ) {
```

Comment:
would it be worth putting a check on the pt fraction of the beam energy (if it is stored anywhere) ?
we have anyway 900 GeV data and Rivet plugins...

Reply:
I'm ignorant - are there really particles with p->mother() == nullptr that are not the initial particles?

Reply:
I think this is true for pruned/packed collections at MINIAOD level

Reply:
yes some intermediate/final state particles loose mother during the MiniAOD pruning.

Reply:
Thanks (though perhaps it should be possible to distinguish true motherless and pruned motherless in the future as to avoid the sort of fragile checks done here)